### ISSUE TITLE
@typescript-eslint/prefer-optional-chain in case of boolean function

### ISSUE THREAD

**Issue by robstoll** [eslint/eslint#15549]
### What rule do you want to change?

prefer-optional-chain

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

Angular project:
```js
  hasRequiredError(form: FormGroup, formFieldName: string): boolean {
    const field = form.get(formFieldName);
    return field != null && field.hasError('required');
    // ^ @typescript-eslint/prefer-optional-chain
  }
```

### What does the rule currently do for this code?

complain

### What will the rule do after it's changed?

should not complain as `field?.hasError('required')` does no longer result in a boolean (but `boolean | undefined, this is OK but not in for cases as above were the result is returned) and `field?.hasError('required') ?? false` doesn't read nicer as the original code

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by ljharb** [comment]
If you're using the typescript-eslint rule, then you need to file an issue on that repo, not this one.

**Comment by robstoll** [comment]
oh... figured it's the same repo by now. sry.
